au/code/au/utility/test/probable_primes_test.cc

```diff
@@ -200,6 +200,52 @@ TEST(MillerRabin, SupportsConstexpr) {
     static_assert(result == PrimeResult::PROBABLY_PRIME, "997 is prime");
 }
 
+TEST(Gcd, ResultIsAlwaysAFactorAndGCDFindsNoLargerFactor) {
+    for (auto i = 0u; i < 500u; ++i) {
+        for (auto j = 1u; j < i; ++j) {
+            const auto g = gcd(i, j);
+            EXPECT_EQ(i % g, 0u);
+            EXPECT_EQ(j % g, 0u);
+
+            // Brute force: no larger factors.
+            for (auto k = g + 1u; k < j / 2u; ++k) {
+                EXPECT_FALSE((i % k == 0u) && (j % k == 0u));
+            }
+        }
+    }
+}
+
+TEST(Gcd, HandlesZeroCorrectly) {
+    // The usual convention: if one argument is 0, return the other argument.
+    EXPECT_EQ(gcd(0u, 0u), 0u);
+    EXPECT_EQ(gcd(10u, 0u), 10u);
+    EXPECT_EQ(gcd(0u, 10u), 10u);
+}
+
+TEST(JacobiSymbol, ZeroWhenCommonFactorExists) {
+    for (int i = -20; i <= 20; ++i) {
+        for (auto j = 1u; j <= 19u; j += 2u) {
+            for (auto factor = 3u; factor < 200u; factor += 2u) {
+                // Make sure that `j * factor` is odd, or else the result is undefined.
+                EXPECT_EQ(jacobi_symbol(i * static_cast<int>(factor), j * factor), 0)
+                    << "jacobi(" << i * static_cast<int>(factor) << ", " << j * factor
+                    << ") should be 0";
+            }
+        }
+    }
+}
+
+TEST(JacobiSymbol, AlwaysOneWhenFirstInputIsOne) {
+    for (auto i = 3u; i < 99u; i += 2u) {
+        EXPECT_EQ(jacobi_symbol(1, i), 1) << "jacobi(1, " << i << ") should be 1";
+    }
+}
+
+TEST(BoolSign, ReturnsCorrectValues) {
+    EXPECT_EQ(bool_sign(true), 1);
+    EXPECT_EQ(bool_sign(false), -1);
+}
+
```

Comment:
Can we get some tests when `jacobi_symbol` is either `-1` or `1`?

Reply:
Done.

While playing around locally, I tested against the C++ snippet on the Wikipedia page.  We reproduced the correct answer for every value in the table.  But I didn't want to check that code in due to questions about licensing a snippet from Wikipedia.

What I've done now is to take specific test cases from the text of the page, each of which is either 1 or -1.